Update index.md

jonvnadelberg opened the pull request:
<!--
Thanks for opening a pull request! Please do not just delete this text.  The three fields below are mandatory.

Please remember to:
- This PR requires an issue. If it is a new feature, the issue should proceed the PR and will have allowed sufficent time for discussions to take place. Please use
issue tags such as "Closes #XYZ" or "Resolves sigstore/repo-name#XYZ".
  [documentation](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)
- ensure your commits are signed-off, as sigstore uses the [DCO](https://en.wikipedia.org/wiki/Developer_Certificate_of_Origin) using `git commit -s`, or `git commit -s --amend` if you want to amend already existing commits
- lastly, ensure there are no merge commits!

Thank you :)
-->

#### Summary

Additional changes for https://github.com/sigstore/docs/issues/168

Clarifying "How to use Sigstore" section.  Adding installation link and spelling out that you use Cosign to both sign and verify.
<!--
 Explain the **motivation** for making this change. What existing problem does the pull request solve? How can reviewers test this PR?
-->

#### Release Note
<!--
Add a release note for each of the following conditions:

* Config changes (additions, deletions, updates)
* API additions—new endpoint, new response fields, or newly accepted request parameters
* Database changes (any)
* Websocket additions or changes
* Anything noteworthy to an administrator running private sigstore instances (err on the side of over-communicating)
* New features and improvements, including behavioural changes, UI changes and CLI changes
* Bug fixes and fixes of previous known issues
* Deprecation warnings, breaking changes, or compatibility notes

If no release notes are required write NONE. Use past-tense.

-->
NONE
#### Documentation
<!--

Does this change require an update to documentation? How will users implement your new feature?

Please reference a PR within https://docs.sigstore.dev

-->
Doc only fix

jonvnadelberg commented:
@olivekl OK, I added a line about the install and changed how verify is mentioned with Cosign.  Please review.

olivekl commented:
Hm, is there a common use case for separating out "install" from "sign and verify?" I.e., most people will install it to do those two things, right? Also, do we want to combine sign and verify into one user journey? I'm not worried about potential signers being confused, but will users who downloaded an artifact and are looking to verify it understand that "sign and verify" is referring to them? @haydentherapper

haydentherapper commented:
Agreed, keeping sign and verify separate is reasonable given we expect different users for each.

haydentherapper commented on content/en/index.md:
```diff
@@ -14,9 +14,9 @@ It’s free to use for all developers and software providers, with Sigstore’s
 * Sign Git commits with [Gitsign](/gitsign/overview/) - [Quick start](/gitsign/overview/#quick-start)
-* Verify entries with [Cosign](/cosign/verify/)
 * Use the [policy controller](/policy-controller/overview/) to enforce Kubernetes policies
 * Use the [Fulcio Certificate Authority](/fulcio/overview/)
```
Can we remove the Fulcio bullet point? I think this section works better if it's focused only on clients.

jonvnadelberg replied:
OK...."and verify" part of the sentence is removed.  @olivekl

ltagliaferri commented on content/en/index.md:
```diff
@@ -14,11 +14,12 @@ It’s free to use for all developers and software providers, with Sigstore’s
 * Use the [policy controller](/policy-controller/overview/) to enforce Kubernetes policies
-* Use the [Fulcio Certificate Authority](/fulcio/overview/)
+
+Once installed, Cosign can also be used to verify.  If you need to verify a signed artifact, install Cosign and follow the  [verify instructions](/cosign/verify/).
```
```suggestion
Once installed, Cosign can also be used to verify.  If you need to verify a signed artifact, install Cosign and follow the [verify instructions](/cosign/verify/).
```

ltagliaferri commented on content/en/index.md:
```diff
@@ -14,11 +14,12 @@ It’s free to use for all developers and software providers, with Sigstore’s
 * Sign Git commits with [Gitsign](/gitsign/overview/) - [Quick start](/gitsign/overview/#quick-start)
-* Verify entries with [Cosign](/cosign/verify/)
 * Use the [policy controller](/policy-controller/overview/) to enforce Kubernetes policies
-* Use the [Fulcio Certificate Authority](/fulcio/overview/)
```
Do we not want Fulcio mentioned at all here?

haydentherapper replied:
My thought was no, for this section. Operating Fulcio/Rekor would fall under developer documentation, rather than user documentation. Maybe a separate section linking to overview pages for Rekor and Fulcio, for "developers or private deployment operators"?

olivekl replied:
Agreed, this section works well for getting end users onto the correct user journey; I don't think we should mix user and developer docs.

jonvnadelberg replied:
I can put in a link to the overviews of Fulcio and Rekor.   Shall we go with that?  or is that mixing user and developer at this point still?

ltagliaferri replied:
I think it may be worthwhile to say that Fulcio is the CA, but we can do that later.

olivekl commented:
> @olivekl OK, I added a line about the install and changed how verify is mentioned with Cosign. Please review.

I'd like to see one line at the top of cosign/verify that says something along the lines of "you'll need to install cosign first" and links to those instructions.

jonvnadelberg commented:
> > @olivekl OK, I added a line about the install and changed how verify is mentioned with Cosign. Please review.
> 
> I'd like to see one line at the top of cosign/verify that says something along the lines of "you'll need to install cosign first" and links to those instructions.

OK, I will add a line to verify.md that says "To  verify, you must install Cosign" or something like that.

ltagliaferri commented:
> > > @olivekl OK, I added a line about the install and changed how verify is mentioned with Cosign. Please review.
> > 
> > 
> > I'd like to see one line at the top of cosign/verify that says something along the lines of "you'll need to install cosign first" and links to those instructions.
> 
> OK, I will add a line to verify.md that says "To verify, you must install Cosign" or something like that.

Will you add this into this PR @jonvnadelberg ?

jonvnadelberg commented:
Yes.   I will do it right now.


On Tue, May 16, 2023 at 7:36 PM ltagliaferri ***@***.***>
wrote:

> @olivekl <https://github.com/olivekl> OK, I added a line about the
> install and changed how verify is mentioned with Cosign. Please review.
>
> I'd like to see one line at the top of cosign/verify that says something
> along the lines of "you'll need to install cosign first" and links to those
> instructions.
>
> OK, I will add a line to verify.md that says "To verify, you must install
> Cosign" or something like that.
>
> Will you add this into this PR @jonvnadelberg
> <https://github.com/jonvnadelberg> ?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/sigstore/docs/pull/170#issuecomment-1550601591>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A5CUIOLQAIONZNWJWSSQJ5DXGQ2TXANCNFSM6AAAAAAX6MLDYQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

olivekl commented on content/en/cosign/verify.md:
```diff
@@ -4,6 +4,8 @@ category: "Cosign"
 position: 106
 ---
 
+> **Note**: If you need to verify a signed artifact or blob, you must first install Cosign. Follow the  [installation instructions](/cosign/installation/) to install.
```
```suggestion
> **Note**: To verify a signed artifact or blob, first [install Cosign](/cosign/installation/), then follow the instructions below.
```

haydentherapper commented on content/en/index.md:
```diff
@@ -14,11 +14,12 @@ It’s free to use for all developers and software providers, with Sigstore’s
 
 ## How to use Sigstore
 
+* Install Cosign - [Cosign Installation page](/cosign/installation/)
```
Good with me. Were we going to link to Verify still?

jonvnadelberg replied:
I put in a line on index.md that says "Once installed, Cosign can also be used to verify. If you need to verify a signed artifact, install Cosign and follow the verify instructions."

jonvnadelberg replied:
I think maybe all these bullet points should go away.   Unless we want to tell people to use the policy controller?   If we want something that says:

I Want to sign something.
I Want to verify something.

with links to the proper places?   Would that be good?

olivekl replied:
I think that's super clear. Would everyone be okay with limiting the "set off on the correct path" options to those two main user journeys?

jonvnadelberg replied:
I changed it a little from the two simple points because you wanting to be able to sign blobs and sign containers are two different things..   I also left in the gitsign thing and putin the I want to verify something.  So the bullet list looks like this now:


I want to [install and Quick Start](https://github.com/jonvnadelberg/sigstore-docs/blob/update-index.md/cosign/overview/#quick-start)
I want to [sign a blob](https://github.com/jonvnadelberg/sigstore-docs/blob/update-index.md/content/en/cosign/signing_with_blobs)
I want to [sign a container](https://github.com/jonvnadelberg/sigstore-docs/blob/update-index.md/content/en/cosign/signing_with_containers)
I want to [Sign Git commits with Gitsign](https://github.com/jonvnadelberg/sigstore-docs/blob/update-index.md/gitsign/overview) - [Quick Start](https://github.com/jonvnadelberg/sigstore-docs/blob/update-index.md/gitsign/overview/#quick-start)
I want to [verify entries with Cosign](https://github.com/jonvnadelberg/sigstore-docs/blob/update-index.md/cosign/verify)


Let me know if more changes are wanted, or we want to go to back to a simpler thing.
